### PR TITLE
feat(content): resolve product, category and content links

### PR DESCRIPTION
**Pull request by supproduction**
Improve the content link component to resolve links to product, category or content pages. 
Reveal suggestions so that integrations can reuse the data while building content. 

closes: [HRZ-89824](https://spryker.atlassian.net/browse/HRZ-89824)

[HRZ-89824]: https://spryker.atlassian.net/browse/HRZ-89824?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/domain/content/link/link.component.ts [hunk 1]
```diff
@@ -21,6 +26,27 @@ export class ContentLinkComponent extends ContentMixin<
     return null;
   });
 
+  protected $text = computed(() => {
+    const { type, id } = this.$options();
```
Comment (tobi-or-not-tobi) [inline libs/domain/content/link/link.component.ts:47]: we could move this down a bit

File: libs/domain/content/link/link.schema.ts [hunk 2]
```diff
@@ -1,75 +1,86 @@
...
+        {
+          value: RouteType.Category,
+          text: 'Category',
+        },
+        {
+          value: RouteType.ProductList,
+          text: 'Search',
+        },
+      ],
+    },
```
Comment (tobi-or-not-tobi) [inline libs/domain/content/link/link.schema.ts]: We could introduce a feature to make those _hidden_ in EB options panel, as for now they're autofilled by the suggestion widget. Or just remove them for now.

File: libs/domain/content/link/link.schema.ts [hunk 3]
```diff
@@ -1,75 +1,86 @@
...
+export const linkComponentSchema: ContentComponentSchema<
+  ContentLinkComponent,
+  { suggestion: unknown }
+> = {
+  name: 'Link',
+  group: 'Content',
+  icon: IconTypes.Link,
+  options: {
+    suggestion: {
+      type: 'suggestion',
```
Comment (tobi-or-not-tobi) [inline libs/domain/content/link/link.schema.ts]: Suggestion sounds to generic. What about ContentSuggestion?

**Review by tobi-or-not-tobi — CHANGES_REQUESTED** [review]

File: libs/domain/content/link/link.schema.ts (hunk 2 above)
Comment (supproduction) [reply libs/domain/content/link/link.schema.ts]: Dropped it now

File: libs/domain/content/link/link.component.ts (hunk 1 above)
Comment (tobi-or-not-tobi) [reply libs/domain/content/link/link.component.ts:47]: nitpick, but still open afaik

**Review by tobi-or-not-tobi — APPROVED** [review]

**Pull request merged**